src/amber/router/pipe/pipeline.cr

```diff
@@ -29,7 +29,7 @@ module Amber
       # Connects pipes to a pipeline to process requests
       def build(valve : Symbol, &block)
         @valve = valve
-        @pipeline[valve] = [] of HTTP::Handler unless pipeline.key? valve
+        @pipeline[valve] = [] of HTTP::Handler unless pipeline.has_key? valve
```

Comment:
This fixes what I think is a bug. `Hash#key?` expects a value as an argument, so this has been re-initializing the `@pipeline[valve]` every time. Essentially over-writing any existing pipes.

Reply:
I incorporated the same change in another branch! Nice Catch